### PR TITLE
relativistic particle

### PR DESCRIPTION
**Pull request by TCLiuu**
1. After lots of tests, I chose Eq.(2) in #34 as the equations of motion.
2. I have rewritten the error message of FTLError. @henry2004y, can you help me review the expression?
3. The purpose of the new test case is to verify whether the energy of electron is the same as expected. So an algorithm with higher accuracy like `Vern6` is necessary.

And, I want to attach a [song](https://www.youtube.com/watch?v=6_eKJysJk1Y).

### CODE REVIEW

**Comment by TCLiuu** [comment]
close #25

File: src/TestParticle.jl [hunk 1]
```diff
@@ -264,13 +279,15 @@ ODE equations for relativistic charged particle moving in static EM field with o
 function trace_relativistic(y, p::TPTuple, t)
    q, m, E, B = p
 
-   if y[4]*y[4] + y[5]*y[5] + y[6]*y[6] ≥ c^2
-      throw(ArgumentError("Particle faster than the speed of light!"))
+   u2 = y[4]^2 + y[5]^2 + y[6]^2
+   if u2 ≥ c^2
```
Comment (henry2004y) [inline src/TestParticle.jl:283]: Same comment as above.

File: test/runtests.jl [hunk 2]
```diff
@@ -211,5 +211,52 @@ end
...
+      u0 = [0.0, 0.0, 0.0] # initial velocity, [m/s]
+      tspan = (0.0, 2e-7)
+      stateinit = [x0..., u0...]
+      param = prepare(E_field, B_field; species=Electron)
+
+      prob = ODEProblem(trace_relativistic!, stateinit, tspan, param)
+      sol = solve(prob, Vern6(); dtmax=1e-10, save_idxs=[1,2,3,4,5,6])
+      x = sol.u[end][1:3]
+
+      @test cal_energy(sol)/(x[1]+x[2]-20) ≈ 1e5
```
Comment (henry2004y) [inline test/runtests.jl:254]: What's been tested here for this expression?

**Review by henry2004y — COMMENTED** [review]
I've went through the changes, but I haven't got time to check the derivations; I trust you for the correctness of equations.

In #34, you mentioned that the accuracy of Eq.(1) is better than Eq.(2), but in this PR you chose to use the later one. What's the reason behind?

File: test/runtests.jl (hunk 2 above)
Comment (henry2004y) [reply test/runtests.jl:254]: 所以
```julia
@test cal_energy(sol)/(x[1]+x[2]-20) ≈ 1e5
```
中的分母指的是电子在X和Y方向的位移距离之和？由于这是和初始位置相关的，我建议改成
```julia
@test cal_energy(sol)/(x[1]-x0[1]+x[2]-x0[2]) ≈ 1e5
```

Comment (TCLiuu) [reply test/runtests.jl]: You are right.

**Pull request merged**